<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="weights" default="$(find-pkg-share rgbd_proc)/config/weights/disp_refine/model.pth" />
  <arg name="bag_path" default="$(find-pkg-share rgbd_proc)/data/bags/helhest/helhest_2025_04_29-16_10_48/" />

  <!-- Play bag file -->
  <include file="$(find-pkg-share rgbd_proc)/launch/helhest_play.launch.py">
    <arg name="bag_path" value="$(var bag_path)" />
  </include>

  <!-- Depth refinement node -->
  <node pkg="rgbd_proc" exec="depth_refinement.py" name="depth_refinement" output="screen">
    <param name="weights" value="$(var weights)" />
    <param name="device" value="cuda" />
    <param name="max_msgs_delay" value="0.05" />
    <param name="max_age" value="0.5" />

    <remap from="/camera_left/image_rect" to="/luxonis/oak/left/image_rect" />
    <remap from="/camera_left/camera_info" to="/luxonis/oak/left/camera_info" />
    <remap from="/camera_right/image_rect" to="/luxonis/oak/right/image_rect" />
    <remap from="/depth" to="/luxonis/oak/stereo/image_raw/depth" />
  </node>

  <!-- RVIZ2 -->
  <node if="0" pkg="rviz2" exec="rviz2" name="rviz2"
      args="-d $(find-pkg-share rgbd_proc)/config/rviz/depth_refinement.rviz"/>

</launch>
